prow/config/jobs/test-infra.yaml

```diff
@@ -12,6 +12,17 @@ jobs:
   - name: gencheck
     command: [make, gen-check]
 
+  - name: unit-test-authentikos
+    type: presubmit
+    regex: '^authentikos/(.+\.go|go\.mod)$'
+    command: [make, -C, authentikos, unit-test]
+
+  - name: integ-test-authentikos
+    type: presubmit
+    regex: '^authentikos/(test/.+|.+\.go|go\.mod)$'
```

Comment:
nit: the regex is fairly complicated, can you comment what we are matching and why?